validator/sawtooth_validator/execution/executor.py

```diff
@@ -77,47 +102,188 @@ def run(self):
+                required_transaction_processors += [config_processor]
+
+            if processor_type not in required_transaction_processors:
+                # If there aren't any configured required transaction
+                # processors, wait for all transactions. This is useful
+                # for startup with the assumption being that the
+                # required transaction processors would be configured soon
+                # after startup.
+                if len(required_transaction_processors) == 0:
+                    # There are no required txn processors
+                    LOGGER.debug("there are no required transaction "
+                                 "processors")
+                    self._execute_or_wait_for_processor_type(
+                        processor_type=processor_type,
+                        content=content)
+                else:
+                    # The txn processor type is not in the required
+                    # transaction processors so
+                    # failing transaction right away
+                    LOGGER.debug("failing transaction of type (name=%s,"
+                                 "version=%s,encoding=%s) since it isn't"
+                                 " required in the configuration",
+                                 processor_type.name,
+                                 processor_type.version,
+                                 processor_type.encoding)
+                    self._context_manager.delete_context(
+                        context_id_list=[context_id])
+                    self._scheduler.set_transaction_execution_result(
+                        txn_signature=txn.header_signature,
+                        is_valid=False,
+                        context_id=context_id)
+            else:
+                # The processor type is a required one
+                # so execute or wait for capabilities
+                self._execute_or_wait_for_processor_type(
+                    processor_type=processor_type,
+                    content=content)
+
+    def _execute_or_wait_for_processor_type(self, processor_type, content):
+        processor = self._processors.get_next_of_type(
+            processor_type=processor_type)
+        if processor is None:
+            LOGGER.debug("no transaction processors registered for "
+                         "processor type %s", processor_type)
+            if processor_type not in self._waiters_by_type:
+                in_queue = queue.Queue()
+                in_queue.put_nowait(content)
+                waiter = _Waiter(self._send_and_process_result,
+                                 processor_type=processor_type,
+                                 processors=self._processors,
+                                 in_queue=in_queue,
+                                 waiters_by_type=self._waiters_by_type)
+                self._waiters_by_type[processor_type] = waiter
+                waiter.start()
```

Comment:
All Thread start()s must have an appropriate matching join() at the right time.  This might be a good reason to use a thread pool here.